cmd/id.go

```diff
@@ -12,7 +13,7 @@ func NewAgentIDCmd() *cobra.Command {
 		Use:   "id",
 		Short: "Print the agent identifier",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			agentID, err := agent.GetAgentID()
+			agentID, err := agent.GetAgentID(afero.NewOsFs())
```

Comment:
This PR opens the need to refactor also `id` command I guess, because as of now it will always get the _real agent id_.

What I would expect is that when `./trento-agent id` is executed it returns either what has been configured in agent's config file or the real agent id.

As an addition I wouldn't expect `./trento-agent id --force-agent-id=blaa`, as it would be fancy :sweat_smile: 

Finally we'd need to take care of having the same defaults for both the commands, otherwise in one case viper would complain about not having some expected flags.

Reply:
I wouldn't go in this direction. This `force-agent-id` is a hidden flag for development/test purposes, and it shouldn't be used together with the `id` command, which is used to get the real id. It is only for the `start` function, which is the one that it makes sense.
Using `./trento-agent id --force-agent-id=blaa` doesn't make any sense.

Reply:
Yeah true, that's just for test/dev so if that's how it has to be I can live with this little inconsistency :wink:

> Using ./trento-agent id --force-agent-id=blaa doesn't make any sense.
>
Yes, we're saying the same about this :smile: